latch/registry/table.py

```diff
@@ -0,0 +1,275 @@
+
+import gql
+
+from latch.gql.execute import execute, get_transport
+from latch.registry.record import Record
+from latch.registry.utils import (
+    clean,
+    to_python_literal,
+    to_python_type,
+    to_registry_literal,
+)
+
+
+class ListRecordsOutputType(TypedDict):
+    records: List[Record]
+    errors: List[Exception]
+
+
+class Table:
+    def __init__(self, id: str):
+        self.id = id
+        self._name: Optional[str] = None
+        self._record_type: Type[Record] = None
+        self._columns: Optional[List[Dict[str, Dict]]] = None
+        self._info: Optional[Dict[str, Dict]] = None
+
+    def load(self):
+        self.get_display_name(load_if_missing=True)
+        self.get_columns(load_if_missing=True)
+
+    def _get_info(self):
+        if self._info is not None:
+            return self._info
+
+        self._info = execute(
+            """
+            query tableQuery ($argTableId: BigInt!) {
+                catalogExperiment(id: $argTableId) {
+                    id
+                    displayName
+                    catalogExperimentColumnDefinitionsByExperimentId {
+                        nodes {
+                            key
+                            type
+                        }
+                    }
+                    catalogSamplesByExperimentId(condition: { removed: false }) {
+                        nodes {
+                            id
+                            name
+                            catalogSampleColumnDataBySampleId {
+                                nodes {
+                                    data
+                                    key
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            """,
+            variables={"argTableId": self.id},
+        )
+
+        return self._info
+
+    def get_display_name(self, load_if_missing=False):
+        if not load_if_missing or self._name is not None:
+            return self._name
+
+        self._name = self._get_info()["catalogExperiment"]["displayName"]
+
+        return self._name
+
+    def get_columns(self, load_if_missing=False):
+        if not load_if_missing or self._columns is not None:
+            return self._columns
+
+        self._columns = self._get_info()["catalogExperiment"][
+            "catalogExperimentColumnDefinitionsByExperimentId"
+        ]["nodes"]
+
+        return self._columns
+
+    def list_records(self) -> Iterator[ListRecordsOutputType]:
+        has_next_page = True
+        end_cursor = None
+
+        while has_next_page:
+            data = execute(
+                """
+                query tableQuery($argTableId: BigInt!, $argAfter: Cursor) {
+                    catalogExperiment(id: $argTableId) {
+                        catalogSamplesByExperimentId(
+                            condition: { removed: false }
+                            first: 50
```

Comment:
```suggestion
                            first: 100
```
probably should be even higher but we don't have any tables that would really benefit from this right now anyway

Reply:
probably take this as a keyword-only `page_size` parameter too